internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -402,6 +422,12 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, bmh *bmh_v1alpha1.Bar
 	r.Log.Debugf("Started BMH reconcile for %s/%s", bmh.Namespace, bmh.Name)
 	r.Log.Debugf("BMH value %v", bmh)
 
+	err := r.ensureMCSCert(ctx, bmh)
+	if err != nil {
+		r.Log.WithError(err).Errorf("failed to get MCS cert")
+		return reconcileError{err}
+	}
+
```

Comment:
What do you think about moving this under L482? 

The reasoning is the following. We are only calling this ensure when the `bmh.Spec.Image.URL` hasn't been set (L418) so it makes sense to only set it when the ISOUrl is already present in the `InfraEnv`. This way, we will avoid calling it on every `reconcileBMH` call and instead we would call it only when there is something to update.

As it is now, we would set the annotation but we won't save it since `return reconcileComplete{true}` is only being returned once the `bmh.Spec.Image.URL` is set.